gossipd/gossip.c

```diff
@@ -1772,6 +1772,85 @@ static struct io_plan *handle_txout_reply(struct io_conn *conn,
 static struct io_plan *handle_disable_channel(struct io_conn *conn,
 					      struct daemon *daemon, u8 *msg)
 {
+	tal_t *tmpctx = tal_tmpctx(msg);
+	struct short_channel_id scid;
+	u8 direction;
+	struct node_connection *nc;
+	bool active;
+	u16 flags, cltv_expiry_delta;
+	u32 timestamp, fee_base_msat, fee_proportional_millionths;
+	struct bitcoin_blkid chain_hash;
+	secp256k1_ecdsa_signature sig;
+	u64 htlc_minimum_msat;
+
+	if (!fromwire_gossip_disable_channel(msg, NULL, &scid, &direction, &active) ) {
+		status_trace("Unable to parse %s",
+			     gossip_wire_type_name(fromwire_peektype(msg)));
+		goto fail;
+	}
+
+	nc = get_connection_by_scid(daemon->rstate, &scid, direction);
+
+	if (!nc) {
+		status_trace(
+		    "Unable to find channel %s/%d",
+		    type_to_string(msg, struct short_channel_id, &scid),
+		    direction);
+		goto fail;
+	}
+
+	status_trace("Disabling channel %s/%d, active %d -> %d",
+		     type_to_string(msg, struct short_channel_id, &scid),
+		     direction, nc->active, active);
+
+	nc->active = active;
+
+	if (!nc->channel_update) {
+		status_trace(
+		    "Channel %s/%d doesn't have a channel_update yet, can't "
+		    "disable",
+		    type_to_string(msg, struct short_channel_id, &scid),
+		    direction);
+		goto fail;
```

Comment:
In this case, what if active was true?  Did we just enable a channel w/o a channel update?

I think this test should be moved up above the assignment?

Reply:
Can't do that, otherwise we can only ever deactivate a channel that has reached `announce_depth`, i.e., 6 blocks. The channel is activated locally after `funding_depth`, i.e., 2 confirmations. So if we happen to lose the connection or similar, which the channel has 2-5 confirmations we cannot deactivate and we're stuck with a non-functioning channel being returned as part of the routes.

If the channel is public, we also have a `channel_update` which we can update with the disable flag, if the channel is not public, then others don't care anyway.

Reply:
Good point.

OK, my q still stands.  We're asserting that active == false in this branch (based on the msg), we should actually assert that?

I'll apply this now, but please consider...